pyiceberg/table/__init__.py

```diff
@@ -539,6 +578,81 @@ def overwrite(
                 for data_file in data_files:
                     update_snapshot.append_data_file(data_file)
 
+    def delete(self, delete_filter: Union[str, BooleanExpression], snapshot_properties: Dict[str, str] = EMPTY_DICT) -> None:
+        """
+        Shorthand for deleting record from a table.
+
```

Comment:
One last minor comment: Shall we also mention that `delete` may produce different types of snapshot operation based on
```
- DELETE: In case existing Parquet files can be dropped completely.
- REPLACE: In case existing Parquet files need to be rewritten
```
like the docstring of `overwrite`?

Reply:
Yes, thanks!